hat/assets/js/apps/Iaso/components/files/pdf/DocumentUploadWithPreview.tsx

```diff
@@ -0,0 +1,55 @@
+import { Grid } from '@mui/material';
+import { FilesUpload, useSafeIntl } from 'bluesquare-components';
+import React from 'react';
+import { PdfPreview } from './PdfPreview';
+import MESSAGES from './messages';
+import { acceptPDF } from './utils';
+
+type DocumentUploadWithPreviewProps = {
+    errors: string[] | undefined;
+    onFilesSelect: (files: File[]) => void;
+    document?: File[] | string;
+};
+
+const DocumentUploadWithPreview: React.FC<DocumentUploadWithPreviewProps> = ({
+    errors,
+    onFilesSelect,
+    document,
+}) => {
+    const { formatMessage } = useSafeIntl();
+
+    let pdfUrl: string | null = null;
+    if (typeof document === 'string') {
+        pdfUrl = document;
+    } else if (
+        Array.isArray(document) &&
+        document.length > 0 &&
+        document[0] instanceof File
+    ) {
+        pdfUrl = URL.createObjectURL(document[0]);
+    } else if (document instanceof File) {
+        pdfUrl = URL.createObjectURL(document[0]);
+    }
```

Comment:
Are you sure this is correct? The 2 `else if` do exactly the same thing

Reply:
you are right